#OpenXT:
#	This patch enables auto dhcp on eth0 and brbridged which is disabled by default.
#	auto dhcp is disabled by default on eth0 as it is a slave to brbridged.
#	auto dhcp is disabled by default on bridge devices.
--- a/src/nm-ip4-config.c
+++ b/src/nm-ip4-config.c
@@ -1055,7 +1055,7 @@ nm_ip4_config_create_setting (const NMIP
 
 	if (!self) {
 		g_object_set (s_ip4,
-		              NM_SETTING_IP_CONFIG_METHOD, NM_SETTING_IP4_CONFIG_METHOD_DISABLED,
+		              NM_SETTING_IP_CONFIG_METHOD, NM_SETTING_IP4_CONFIG_METHOD_AUTO,
 		              NULL);
 		return NM_SETTING (s_ip4);
 	}
@@ -1100,7 +1100,7 @@ nm_ip4_config_create_setting (const NMIP
 
 	/* Use 'disabled' if the method wasn't previously set */
 	if (!method)
-		method = NM_SETTING_IP4_CONFIG_METHOD_DISABLED;
+		method = NM_SETTING_IP4_CONFIG_METHOD_AUTO;
 
 	g_object_set (s_ip4,
 	              NM_SETTING_IP_CONFIG_METHOD, method,
--- a/src/devices/nm-device.c
+++ b/src/devices/nm-device.c
@@ -9976,10 +9976,12 @@ nm_device_activate_stage3_ip4_start (NMD
 	NMActStageReturn ret;
 	NMDeviceStateReason failure_reason = NM_DEVICE_STATE_REASON_NONE;
 	gs_unref_object NMIP4Config *ip4_config = NULL;
+	char *iface = NULL;
 
 	g_assert (priv->ip_state_4 == NM_DEVICE_IP_STATE_WAIT);
+	iface = nm_device_get_ip_iface (self);
 
-	if (nm_device_sys_iface_state_is_external (self)) {
+	if (nm_device_sys_iface_state_is_external (self) && strncmp (iface, "brbridged", 9)) {
 		_set_ip_state (self, AF_INET, NM_DEVICE_IP_STATE_DONE);
 		check_ip_state (self, FALSE, TRUE);
 		return TRUE;
@@ -14877,6 +14879,7 @@ _set_state_full (NMDevice *self,
 	gboolean no_firmware = FALSE;
 	NMSettingsConnection *sett_conn;
 	NMSettingSriov *s_sriov;
+	char *iface = NULL;
 
 	g_return_if_fail (NM_IS_DEVICE (self));
 
@@ -14923,9 +14926,11 @@ _set_state_full (NMDevice *self,
 	/* Cache the activation request for the dispatcher */
 	req = nm_g_object_ref (priv->act_request.obj);
 
+	iface = nm_device_get_ip_iface (self);
 	if (   state >  NM_DEVICE_STATE_UNMANAGED
 	    && state <= NM_DEVICE_STATE_ACTIVATED
-	    && nm_device_state_reason_check (reason) == NM_DEVICE_STATE_REASON_NOW_MANAGED
+	    && (nm_device_state_reason_check (reason) == NM_DEVICE_STATE_REASON_NOW_MANAGED
+	    || !strncmp (iface, "brbridged", 9))
 	    && NM_IN_SET_TYPED (NMDeviceSysIfaceState,
 	                        priv->sys_iface_state,
 	                        NM_DEVICE_SYS_IFACE_STATE_EXTERNAL,
